Android/app/src/main/java/com/example/boardgamesocial/MainApp/Fragments/HostedGamesFragment.java

```diff
@@ -166,4 +192,31 @@ private Integer getSeatsAvailableByGameId(Integer gameId) {
         else
             return 0;
     }
+
+    //Method added for updating hostedGames in an event
+    private Observable<JsonArray> fetchUpdatedEventHostedGames() {
+
+        Log.i(TAG, "Inside fetchUpdatedEventHostedGames");
+
+        //Retrieve the value
+        assert getArguments() != null;
+        event = (Event) getArguments().getSerializable(EventVH.EVENT_KEY);
+
+        //Fetching all games in the specified event
+        RetrofitClient.getClient().getCall(HostedGame.class, new HashMap<String, String>() {{
+            put("eventId", event.getId().toString());
+        }}).blockingSubscribe(hosted -> hostedGameList = getObjectList(hosted, HostedGame.class));
+
+        JsonArray jsonArray = new JsonArray();
+
+        //Fetching game details for game ids
+        hostedGameList.forEach(hostedGame -> RetrofitClient.getClient().getCall(Game.class, new HashMap<String, String>() {{
+            put("id", hostedGame.getGameId().toString());
+        }}).blockingSubscribe(j -> {
+            if(!j.isEmpty())
+                jsonArray.add(j.get(0));
+        }));
```

Comment:
You might want to merge the observables before you subscribe to them

Reply:
I see, since observables is a new thing, would have a look into it and shall try to fix it with a new pr. Thanks, I think this is probably the part that may also fix the game refresh issue